### ISSUE TITLE
Tests in the engine repo should output an HTML test report

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#143470]
For accessibility reasons, and honestly, sanity reasons, we need a way to view test failures/success outside of plain text logs.

A starting point might be a small Dart fork of https://github.com/burlachenkok/gtest_report that we emit as an artifact[^1] on LUCI.

[^1]: I asked @zanderso how that part might be possible, and he pointed me to `FLUTTER_LOGS_DIR` (see https://github.com/flutter/engine/blob/main/ci/builders/README.md#tests), of which an example of using this log directory is [`binary_Size_treemap.sh`](https://github.com/flutter/engine/blob/main/ci/builders/linux_android_aot_engine.json#L121).
